### PR TITLE
[GH-443] [Dungeon] [Config] Create DungeonSettlementLevel config json import

### PR DESCRIPTION
**Pull request by lotuuu**
## Motivation

Add DungeonSettlementLevel config json import
Closes https://github.com/lambdaclass/afk_gacha_game/issues/443

## Summary of changes

- Create json with data from https://docs.google.com/spreadsheets/d/12nSgWtkOFmL_iTCDm2LmL3DW5GAgF2jW4fqfiQJRoPs/edit#gid=2012627134
- Create import function using new upsert calls
- Change rates so that they are daily instead of second-ly 

## How to test it?

Run `Champions.Config.import_dungeon_settlement_levels_config()` (Seeds run it automatically). Check that all data is correctly written to DB and that level up works. If you want to test level up with client, you can do so with [this](https://github.com/lambdaclass/afk_gacha_game/pull/472) PR.

### CODE REVIEW

File: apps/champions/lib/champions/config.ex [hunk 1]
```diff
@@ -133,4 +132,38 @@ defmodule Champions.Config do
       )
     end)
   end
+
+  @seconds_in_day 86_400
```
Comment (agustinesco) [inline apps/champions/lib/champions/config.ex:136]: I think this should be at the beginning of the file

File: apps/champions/lib/champions/config.ex [hunk 2]
```diff
@@ -133,4 +132,38 @@ defmodule Champions.Config do
...
+
+  def import_dungeon_settlement_levels_config() do
+    {:ok, dungeon_settlement_levels_json} =
+      Application.app_dir(:champions, "priv/dungeon_settlement_levels.json")
+      |> File.read()
+
+    supplies =
+      Users.Currencies.get_currency_by_name_and_game!("Supplies", Utils.get_game_id(:champions_of_mirra))
```
Comment (agustinesco) [inline apps/champions/lib/champions/config.ex:144]: Should this currency be in the config file too?

File: apps/game_backend/lib/game_backend/users.ex [hunk 3]
```diff
@@ -211,7 +211,48 @@ defmodule GameBackend.Users do
       nil
   """
   def get_dungeon_settlement_level(level_number) do
```
Comment (agustinesco) [inline apps/game_backend/lib/game_backend/users.ex:213]: I'm not sure if the module users is the one for this kind of operations, specially when this isn't related with the user and this belongs to champions. What do you think?

**Review by agustinesco — CHANGES_REQUESTED** [review]
I was check the DungeonSettlementLevels inserted in the seeds and i think there's something off:

```
%GameBackend.Users.DungeonSettlementLevel{
    __meta__: #Ecto.Schema.Metadata<:loaded, "dungeon_settlement_levels">,
    id: "22ff2b14-67e2-4184-bc19-b8324d8410f3",
    level: 1,
    max_dungeon: 25,
    max_factional: 0,
    supply_limit: 10,
    afk_reward_rates: [
      %GameBackend.Campaigns.Rewards.AfkRewardRate{
        __meta__: #Ecto.Schema.Metadata<:loaded, "afk_reward_rates">,
        id: "c4f7633f-f6cb-454a-b0e3-925dddec72ec",
        kaline_tree_level_id: nil,
        kaline_tree_level: #Ecto.Association.NotLoaded<association :kaline_tree_level is not loaded>,
        dungeon_settlement_level_id: "22ff2b14-67e2-4184-bc19-b8324d8410f3",
        dungeon_settlement_level: #Ecto.Association.NotLoaded<association :dungeon_settlement_level is not loaded>,
        currency_id: "2366ef07-e998-40b3-a883-cd3ef86940d9",
        currency: #Ecto.Association.NotLoaded<association :currency is not loaded>,
        rate: 8.333333333333334e-4,
        inserted_at: ~N[2024-05-31 22:30:02],
        updated_at: ~N[2024-05-31 22:30:02]
      }
    ],
    level_up_costs: [
      %GameBackend.Users.Currencies.CurrencyCost{
        currency_id: "3d389684-fc22-4f38-b517-6e1f7b8e4fcf",
        currency: #Ecto.Association.NotLoaded<association :currency is not loaded>,
        amount: 2
      },
      %GameBackend.Users.Currencies.CurrencyCost{
        currency_id: "c5be1961-8569-41d7-b397-be79b8eb203e",
        currency: #Ecto.Association.NotLoaded<association :currency is not loaded>,
        amount: 200000
      }
    ],
    inserted_at: ~N[2024-05-31 22:30:02],
    updated_at: ~N[2024-05-31 22:30:02]
  },
```

Is this rate okay? `rate: 8.333333333333334e-4,`
also are they supposed to have a kaline_tree_level_id?

File: apps/champions/lib/champions/config.ex (hunk 2 above)
Comment (lotuuu) [reply apps/champions/lib/champions/config.ex:144]: I will check with @marco-paulucci!

File: apps/champions/lib/champions/config.ex (hunk 1 above)
Comment (lotuuu) [reply apps/champions/lib/champions/config.ex:136]: It's only used for the function below it, why would you move it to the top of the file 🤔 ? You gain nothing + you have to scroll all the way back up to check the value

**Comment by lotuuu** [comment]
What's off there? The rate is fine, yes. Rates are rather low for Supplies. [You get one every 19-20 minutes](https://docs.google.com/spreadsheets/d/12nSgWtkOFmL_iTCDm2LmL3DW5GAgF2jW4fqfiQJRoPs/edit#gid=2012627134). Since the game design has given us the amount for the rate in the shape of "Currency per day", it's safe to assume that's how they want it (as opposed to needing to put that super low number) for convenience's sake.

If you're afraid of the number, you can check for yourself that it's fine:
```
72 supplies per day / 24 hours in day = 3 supplies per hour
3 supplies per hour / 60 minutes in hour = 0,05 supplies per minute
0,05 supplies per minute / 60 seconds in minute = 0,00083333 supplies per second
```
Which is then simplified to 
`72 / (24 * 60 * 60) = 72 / 86.400 = 0,00083333`

---

No they are not supposed to have a kaline_tree_level_id. I've expanded the AfkRewardRate moduledoc a bit. Please let me know if you find the explanation clear or if it needs more detail!

**Comment by lotuuu** [comment]
About the rate, after talking with @marco-paulucci we have decided to refactor the code slightly so that rates are expressed per day instead of per second, so as to keep it homogeneous between backend & design. I'll be making said refactor here as it should be a low amount of code that's also already tested in our suite, and changes a couple of details in this pr's implementation.

**Comment by agustinesco** [comment]
yeah i found that the rate number was weird but your calculations are right, idk why it had that format. Nice change! i'd rather have a human readable rate and calculate that when we want to give rewards.

On a side note, maybe the "rate" name is a little bit abstract, what about change it to reward_per_day_rate or something like that? not mandatory tho

**Review by agustinesco — APPROVED** [review]
LGTM!

**Comment by lotuuu** [comment]
mmhh yeah I see no issues with renaming rate to daily_rate or something, since that calculation is not per-game should we want to implement it for another one in the future - its' calculation goes on in GameBackend. I'd like to know what @ncontinanza's take on this, I _think_ he was the one who named it initially.

**Comment by ncontinanza** [comment]
I like `daily_rate` better, it clarifies what we're trying to represent with it and we don't depend on documenting or remembering what time unit we're using for the rates.

**Review by ncontinanza — APPROVED** [review]
LGTM!

**Review by agustinesco — APPROVED** [review]
LGTM!

**Pull request merged**